third_party/stdlib/test/test_sched.py

```diff
@@ -0,0 +1,166 @@
+
+TIMEOUT = 10
+
+
+class Timer(object):
+    def __init__(self):
+        self._cond = threading.Condition()
+        self._time = 0
+        self._stop = 0
+
+    def time(self):
+        with self._cond:
+            return self._time
+
+    # increase the time but not beyond the established limit
+    def sleep(self, t):
+        assert t >= 0
+        with self._cond:
+            t += self._time
+            while self._stop < t:
+                self._time = self._stop
+                self._cond.wait()
+            self._time = t
+
+    # advance time limit for user code
+    def advance(self, t):
+        assert t >= 0
+        with self._cond:
+            self._stop += t
+            self._cond.notify_all()
+
+
+class TestCase(unittest.TestCase):
+
+    def test_enter(self):
+        l = []
+        fun = lambda x: l.append(x)
+        scheduler = sched.scheduler(time.time, time.sleep)
+        for x in [0.5, 0.4, 0.3, 0.2, 0.1]:
+            z = scheduler.enter(x, 1, fun, (x,))
+        scheduler.run()
+        self.assertEqual(l, [0.1, 0.2, 0.3, 0.4, 0.5])
+
+    def test_enterabs(self):
+        l = []
+        fun = lambda x: l.append(x)
+        scheduler = sched.scheduler(time.time, time.sleep)
+        for x in [0.05, 0.04, 0.03, 0.02, 0.01]:
+            z = scheduler.enterabs(x, 1, fun, (x,))
+        scheduler.run()
+        self.assertEqual(l, [0.01, 0.02, 0.03, 0.04, 0.05])
+
+    #@unittest.skipUnless(threading, 'Threading required for this test.')
+    #def test_enter_concurrent(self):
+    #    q = queue.Queue()
+    #    fun = q.put
+    #    timer = Timer()
+    #    scheduler = sched.scheduler(timer.time, timer.sleep)
+    #    scheduler.enter(1, 1, fun, (1,))
+    #    scheduler.enter(3, 1, fun, (3,))
+    #    t = threading.Thread(target=scheduler.run)
+    #    t.start()
+    #    timer.advance(1)
+    #    self.assertEqual(q.get(timeout=TIMEOUT), 1)
+    #    self.assertTrue(q.empty())
+    #    for x in [4, 5, 2]:
+    #        z = scheduler.enter(x - 1, 1, fun, (x,))
+    #    timer.advance(2)
+    #    self.assertEqual(q.get(timeout=TIMEOUT), 2)
+    #    self.assertEqual(q.get(timeout=TIMEOUT), 3)
+    #    self.assertTrue(q.empty())
+    #    timer.advance(1)
+    #    self.assertEqual(q.get(timeout=TIMEOUT), 4)
+    #    self.assertTrue(q.empty())
+    #    timer.advance(1)
+    #    self.assertEqual(q.get(timeout=TIMEOUT), 5)
+    #    self.assertTrue(q.empty())
+    #    timer.advance(1000)
+    #    t.join(timeout=TIMEOUT)
+    #    self.assertFalse(t.is_alive())
+    #    self.assertTrue(q.empty())
+    #    self.assertEqual(timer.time(), 5)
+
+    def test_priority(self):
+        l = []
+        fun = lambda x: l.append(x)
+        scheduler = sched.scheduler(time.time, time.sleep)
+        for priority in [1, 2, 3, 4, 5]:
+            z = scheduler.enterabs(0.01, priority, fun, (priority,))
+        scheduler.run()
+        self.assertEqual(l, [1, 2, 3, 4, 5])
+
+    #def test_cancel(self):
```

Comment:
Can you mark these with `@unittest.skip('grumpy')` instead of commenting these tests out?